lib/primitives/keyring.js

```diff
@@ -154,19 +155,26 @@ class KeyRing {
    * Inject data from public key.
    * @private
    * @param {Buffer} key
+   * @param {Boolean?} compress
+   * @returns {KeyRing}
    */
 
-  fromPublic(key) {
+  fromPublic(key, compress) {
     assert(Buffer.isBuffer(key), 'Public key must be a buffer.');
     assert(secp256k1.publicKeyVerify(key), 'Not a valid public key.');
-    this.publicKey = key;
+
+    if (typeof compress === 'boolean')
+      this.publicKey = secp256k1.publicKeyConvert(key, compress !== false);
```

Comment:
Can you get rid of the double negative here please? Factoring out the negation,
`secp256k1.publicKeyConvert(key, compress !== false);` is equivalent to
`secp256k1.publicKeyConvert(key, compress);`